src/components/Contacts/ContactDetails/ContactDetailContext.tsx

```diff
@@ -76,6 +78,37 @@ export const ContactDetailProvider: React.FC<Props> = ({ children }) => {
 
   const [anchorEl, setAnchorEl] = useState<EventTarget & HTMLButtonElement>();
 
+  useEffect(() => {
+    const personId = router.query.personId as string;
+    if (personId) {
+      setSelectedTabKey(TabKey.ContactDetails);
+      setEditPersonModalOpen(personId);
+    }
+  }, [router.query.personId]);
+
+  useEffect(() => {
+    if (!router.query.personId && editPersonModalOpen) {
+      setEditPersonModalOpen(undefined);
+    }
+  }, [router.query.personId]);
+
+  const openPersonModal = (id: string) => {
+    router.replace(
+      { pathname: router.pathname, query: { ...router.query, personId: id } },
+      undefined,
+      { shallow: true },
+    );
+    setEditPersonModalOpen(id);
+  };
+
+  const closePersonModal = () => {
+    const { personId: _, ...rest } = router.query;
+    router.replace({ pathname: router.pathname, query: rest }, undefined, {
+      shallow: true,
+    });
+    setEditPersonModalOpen(undefined);
+  };
```

Comment:
Wrap these in useMemo functions as functions get recreated every render, which will cause everything under the contactDetailContext to render al the time

Reply:
Sorry, useCallback is better. Thank you for changing